cranelift/codegen/src/ir/immediates.rs

```diff
@@ -479,7 +479,7 @@ impl FromStr for Offset32 {
 /// [PartialOrd] respects IEEE754 semantics.
 ///
 /// All bit patterns are allowed.
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+#[derive(Copy, Clone, Debug, Eq, Ord, PartialEq, Hash)]
 #[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
 #[repr(C)]
 pub struct Ieee32(u32);
```

Comment:
I'm concerned about marking `Ieee32` and `Ieee64` as `Ord`. Their `PartialOrd` implementation really is partial, since it's based on IEEE754 ordering. We might need to split these into two versions: one that respects IEEE754 except for preserving exact NaN representations, and another where ordering and equality are defined entirely by bitwise representation. I had a really hard time balancing those two competing goals recently.

Reply:
Removed! This was needed for an earlier egraph implementation but no longer used, so I've removed all of the now-extraneous `Ord` derivations on IR types.